Cython/Includes/libcpp/map.pxd

```diff
@@ -7,32 +7,58 @@ cdef extern from "<map>" namespace "std" nogil:
         ctypedef pair[const T, U] value_type
         ctypedef COMPARE key_compare
         ctypedef ALLOCATOR allocator_type
+
+        cppclass const_iterator
         cppclass iterator:
+            iterator() except +
+            iterator(iterator&) except +
+            # correct would be value_type& but this does not work
+            # well with cython's code gen
             pair[T, U]& operator*()
```

Comment:
I am not super happy with this, but this appears to be an already existing work around so it plays well with cythons code gen. It appears cython derives the return type:
`std::map<T, U>::value_type` even though it actually returns the reference `std::map<T, U>::value_type&`.

Reply:
So … should we fix the type inference/analysis for this?

Reply:
This would probably make sense, but is outside the scope of this PR. I will open another issue for this.

Reply:
Thinking about this again I am unsure whether we can really do better. `std::pair<const T, U>` can not be reassigned to. However this is required in out case, since we define the variables at the top of the function and only assign the value later.